ambry-store/src/main/java/com.github.ambry.store/DiskIOScheduler.java

```diff
@@ -0,0 +1,59 @@
+package com.github.ambry.store;
+
+import com.github.ambry.utils.Throttler;
+import java.io.Closeable;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+
+/**
+ * Helps schedule I/O operations.
+ * This is meant to be used by
+ * 1. Application reads/writes from/to the log.
+ * 2. Hard delete
+ * 3. Compaction
+ * The initial implementation simply returns MAX_VALUE for the application and use the throttler for the other 2.
+ * In the future this will have functions to submit feedback so that more intelligent decisions can be made.
+ */
+class DiskIOScheduler implements Closeable {
+  private final Map<String, Throttler> throttlers = new ConcurrentHashMap<>();
+
+  /**
+   * Create a {@link DiskIOScheduler}.
+   * @param throttlers the {@link Throttler}s to use for each job type.
+   */
+  DiskIOScheduler(Map<String, ? extends Throttler> throttlers) {
```

Comment:
Yes, right now this is kind of a dummy implementation. For now, it is just being passed into the stores for future use. The idea is that it will use the correct throttler for a job type and then return a value that represents the amount of I/O that something should use for ideal disk performance. For now, this value is just MAX_LONG, but in the future it might do something intelligent.